hack/environment.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+get_cd_registry () {
+  # for creating and uploading a component descriptor just insert the location to your registry e.g. eu.gcr.io/.../cnudie/gardener/development
+  registry=""
+
+  if [ -n "$registry" ]; then
+    echo $registry
+  else
+    info "Please insert your registry for the component descriptors in function get_cd_registry in environment.sh"
+    exit 1
+  fi
+}
+
+get_cd_component_name () {
+  echo "github.com/gardener/gardener-extension-networking-calico"
+}
+
+get_image_registry () {
+  echo "eu.gcr.io/gardener-project/gardener/extensions"
```

Comment:
I am a bit confused now. The usual image registry as defined in https://github.com/gardener/gardener-extension-networking-calico/blob/558fbe3ee1918e768f01d4a38d99c3d5d140f570/Makefile#L17 is
`eu.gcr.io/gardener-project/extensions`
while you use
`eu.gcr.io/gardener-project/gardener/extensions`.
Why is there an additional `gardener` in there? Can we simply reuse the existing registry?
In case we could use just one registry I would vote to unify the docker build and simply always give the image another tag, i.e. the resulting image always get VERSION, latest and VERSION-git-hash (=EFFECTIVE_VERSION).

What do you think?

Reply:
I do not know the repository eu.gcr.io/gardener-project/gardener/extensions. This is not the one used when e.g. a release is made. Therefore this one is used eu.gcr.io/gardener-project/gardener/extensions. You find this repo also configured in .ci/pipeline_definitions. 

I do not know if we could unify the image repo as I do not know where eu.gcr.io/gardener-project/extensions is used. With respect to tagging I'm not sure if we should always add the tags like VERSION and latest. The goal of the new make targets is to create a consistent component descriptor with images for testing purposes which should not interfere with other use cases. Not sure if some other component relies on the version or latest tag.

Reply:
Thanks to your pull request we figured out that the tag used in the makefile was inconsistent to the one used in the continuous integration pipeline. Thanks for that.
The result is that we updated the makefile, which means you should rebase. Furthermore, now the base tag is always the same. The only thing different is the version.
I would vote for removing `CNUDIE_IMAGE_REGISTRY` from the makefile and simply always building with all tags, i.e. short version, effective version (including git has) and `latest`.

Reply:
I'm not sure if we should add short version and latest. The cnudie build targets are just for local Landscaper dev scenarios and we want to reduce the possibility to interfere with other use cases. Especially adding the latest tag might be quite risky, e.g. if you have a local buggy test commit not pushed to origin/master.

Reply:
I do not really get it. If this change is only to support the local landscaper development scenario why would it even matter if there are multiple container tags? You would only use the one with the effective version, i.e. git hash.
I really do not like duplicating things as I fear that it will diverge over time. Therefore, I vote for keeping just one docker build command.

@DockToFuture What do you think?

Reply:
I think you are right. I thought that all tags will be added when pushed to the registry but this is not the case. I have removed the additional build target and added the effective version tag.